### ISSUE TITLE
`RuntimeInformation.ProcessArchitecture` returns X64 architecture for Arm64 tests

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#7013]
### .NET version

7.0

### Did it work in .NET Framework?

No

### Issue description

Tests are run for x86, x64 and arm64 architectures. Some tests could be flaky on a specific architecture, so we introduced an attribute to skip a test when run on a certain architecture (https://github.com/dotnet/winforms/pull/6599). The attribute uses the `System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture` property to determine the architecture. However, this property returns `Architecture.X64` for arm64 tests, which makes skipping tests for arm64 not working as expected.

For example, while `DataGridViewTests.DataGridView_ColumnHeadersHeight_SetWithHandle_GetReturnsExpected` has following attribute:
```
[ConditionalWinFormsTheory(UnsupportedArchitecture = Architecture.Arm64,
    Skip = "Flaky tests, see: https://github.com/dotnet/winforms/issues/6597")]
```
it is still executed on arm64:
![image](https://user-images.githubusercontent.com/102954094/163205970-c8476ec9-21cf-4dc9-84e4-10c421321fca.png)

### Steps to reproduce

1. Add Arm64 as an unsupported architecture on a unit test.
2. Search for the test in CI pipeline test results.

**Comment by RussKie** [comment]
Tests on ARM64 are run under x64.

**Comment by dreddy-work** [comment]
> Tests on ARM64 are run under x64.

@RussKie do you know why we didn't opt to run native on arm64?

**Comment by RussKie** [comment]
AFAIK we don't have arm vms. @mmitche or @MattGal  may correct me though.

**Comment by MattGal** [comment]
Correct; ARM64 VMs exist, but we have yet to receive any / enough quota to utilize them in Azure, so the only ARM64 machines we have are Helix test machines which are all physical hardware.

We have https://github.com/dotnet/core-eng/issues/15539 which tracked trying to help Winforms move their tests to Helix, which would allow ARM64 test execution (despite test building still being on AMD64) but AFAIK that was stalled due to not having someone available to work through or disable the failing tests in Helix, so currently (my understanding is that) Winforms has no actual ARM64 tests running today except maybe on people's personal machines.

**Comment by JeremyKuhne** [comment]
@RussKie, @dreddy-work if tests on the ARM leg are run on x64, what is the value? How are they any different than the x64 leg?
